### PR TITLE
Journal of a Researcher not found in a Qualis feature implementation and tests

### PR DESCRIPTION
**Pull request by Trakton**
In order to see the workflow of the feature, run the GUI tests, or manually do:

1. Populate the database with some stub articles
2. Create a researcher having many of the created articles
3. Create one qualis avaliation for each different article journal you've set
4. Create a qualis having almost every qualis avaliation you've created, but leave some of them missing
5. Create a researcher score using the researcher and qualis previously created

Then, the researcher will be classified according to the qualis avaliations for each journal he has an article in. But for each qualis avaliation you've left missing, the qualis will have no way to score the researcher article based on it's journal. It will them count as a "not avaliated" article, and every not avaliated article will be shown in a list.

---
This change is [Reviewable](https://reviewable.io/reviews/pauloborba/rpa/3)

### CODE REVIEW

File: test/cucumber/steps/JournalNotFoundSteps.groovy [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import pages.CreateQualisAvaliationPage
+import pages.CreateQualisPage
+import pages.ShowResearcherScorePage
+import pages.ListResearcherScorePage
+
+this.metaClass.mixin(cucumber.api.groovy.Hooks)
+this.metaClass.mixin(cucumber.api.groovy.EN)
+
+ResearcherScore score
+
```
Comment (dgmneto) [inline test/cucumber/steps/JournalNotFoundSteps.groovy]: Faltou a implementação de métodos para evitar a repetição de código

File: test/functional/pages/ShowResearcherScorePage.groovy [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+class ShowResearcherScorePage extends Page {
+    static url = "/RPA/researcherScore/show/3/"
+
+    static at = {
+        title ==~ /Ver ResearcherScore/
+    }
+
+    boolean Showing(){
+        $("a", text:"Not Avaliated: 1;")
```
Comment (dgmneto) [inline test/functional/pages/ShowResearcherScorePage.groovy:11]: Essa interação parece depender da linguagem do sistema.

**Review by dgmneto — APPROVED** [review]

File: test/cucumber/steps/JournalNotFoundSteps.groovy [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+}
+When(~/^I click to show the avaliation of researcher "([^"]*)"$/) { String cpf ->
+    page.Click(Researcher.findByCpf(cpf).name)
+}
+Then(~/^I am at the description page for the avaliation$/) { ->
+    at ShowResearcherScorePage
+}
+And(~/^I can see that "([^"]*)" journal wasn't scored$/) { String arg1 ->
+    page.Listing(arg1)
+}
```
Comment (dgmneto) [inline test/cucumber/steps/JournalNotFoundSteps.groovy:169]: Faltou a implementação de métodos para a evitar a repetição de código.
Ex.:
Um helper para a criação de Article
Um helper para a verifição se um certo artigo está presente na lista gerada pelo sistema

File: grails-app/controllers/rpa/ResearcherScoreController.groovy [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+
+        researcherScoreInstance.save flush:true
+
+        request.withFormat {
+            form multipartForm {
+                flash.message = message(code: 'default.updated.message', args: [message(code: 'ResearcherScore.label', default: 'ResearcherScore'), researcherScoreInstance.id])
+                redirect researcherScoreInstance
+                researcherScoreInstance.score = CalculateScore(researcherScoreInstance)
+                researcherScoreInstance.articlesNotFound = NotAvaliated(researcherScoreInstance.researcher.articles, researcherScoreInstance.qualis.avaliations)
```
Comment (dgmneto) [inline grails-app/controllers/rpa/ResearcherScoreController.groovy:74]: Esse trecho poderia ser extraído em um novo método, pois ele é uma repetição do que se passa em save

File: grails-app/controllers/rpa/ResearcherScoreController.groovy [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    /*
+    *   Receives a ReseacherScore from the .gsp view containing a Reseacher and a Qualis
+    *   and scanf throught every Article in the Researcher, counting the score their journal
+    *   has in the Qualis. Returns a String for the view to show the final score.
+    */
+    String CalculateScore(ResearcherScore researcherScoreInstance){
+        Set<Article> researcherArticles = researcherScoreInstance.researcher.articles;
+        Set<QualisAvaliation> qualisAvaliations =  researcherScoreInstance.qualis.avaliations;
```
Comment (dgmneto) [inline grails-app/controllers/rpa/ResearcherScoreController.groovy]: Essas dua última linhas tem uma depêndencia de uma classe além da classe vizinha.

**Comment by pauloborba** [comment]
Reviewed 29 of 37 files at r1, 3 of 19 files at r5.
Review status: 32 of 47 files reviewed at latest revision, 25 unresolved discussions.

---

_[grails-app/controllers/rpa/ResearcherScoreController.groovy, line 129 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWryR3rUVKR7UcAkUcJ:-KWryR3rUVKR7UcAkUcK:b-u129mp) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/controllers/rpa/ResearcherScoreController.groovy#L129)):_

> ``` Groovy
>                 }
>             }
>         }
> ```

merece extrair métodos e deixar a intenção de cada parte mais clara

---

_[grails-app/domain/rpa/Article.groovy, line 6 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrygPUcz4NhZv2rFgA:-KWrygPUcz4NhZv2rFgB:b-4adbbv) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/Article.groovy#L6)):_

> ``` Groovy
>     String tittle
>     String journal
>     String issn
> ```

seria importante ter os autores na próxima iteração

---

_[grails-app/domain/rpa/Article.groovy, line 14 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWryoFikCifIEBdfNJe:-KWryoFikCifIEBdfNJf:bkelxhz) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/Article.groovy#L14)):_

> ``` Groovy
>     }
> 
>     Article(String title, String journal, String issn){
> ```

nao precisa desse tipo de construtor em groovy, já é provido

---

_[grails-app/domain/rpa/Qualis.groovy, line 4 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrz2U6sm8JWz5JZj3i:-KWrz2U6sm8JWz5JZj3j:b-1yf9io) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/Qualis.groovy#L4)):_

> ``` Groovy
> 
> class Qualis {
>     String year
> ```

melhor renomear para descrição na próxima iteração. ano é uma das possíveis descrições 

---

_[grails-app/domain/rpa/QualisAvaliation.groovy, line 3 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrzEWu6Bvq392BYBRw:-KWrzEWu6Bvq392BYBRx:bosryly) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/QualisAvaliation.groovy#L3)):_

> ``` Groovy
> package rpa
> 
> class QualisAvaliation {
> ```

evaluation, não avaliation. renomear tudo

---

_[grails-app/domain/rpa/ResearcherScore.groovy, line 3 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrzdXmRO2aOF94GVqU:-KWrzdXmRO2aOF94GVqV:bhedlia) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/ResearcherScore.groovy#L3)):_

> ``` Groovy
> package rpa
> 
> class ResearcherScore {
> ```

a principio, não precisaria desse conceito... a ideia é armazenar as avaliações feitas? para não ter que gastar tempo fazendo de novo? mesmo assim nao estaria ok, ja que as publicacoes de um pesquisador podem mudar. 

---

_[test/cucumber/JournalNotFound.feature, line 11 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrpjAO7FhPOoW1Z3S0:-KWrpjAO7FhPOoW1Z3S1:bq2g5q1) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L11)):_

> ``` Cucumber
>       And the researcher of cpf "12121212121" has only two articles: "Software Engineerig" published at "NTI" and "Software Development" published at "Nature"
>       When I ask for the score of cpf "12121212121" in the qualis "2012"
>       Then The system creates a string saying that "1" article wasn't scored because it's jornal wasn't found
> ```

it's -> its

seria interessante colocar, para a segunda iteracao, mais Ands apos o Then verificando outros detalhes que devem aparecer além dessa mensagem

---

_[test/cucumber/JournalNotFound.feature, line 14 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrq75DVnEcBYp21DCO:-KWrq75DVnEcBYp21DCP:bowngsz) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L14)):_

> ``` Cucumber
> 
>     Scenario: to list the journal of a researcher article that wasn't found in qualis
>       Given The qualis "2013" has avaliations no avaliation for the journal "Nature"
> ```

has no evaluations 

evaluation não

---

_[test/cucumber/JournalNotFound.feature, line 16 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrr1l1clVdZ20tdNCY:-KWrr1l1clVdZ20tdNCZ:b9iqwip) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L16)):_

> ``` Cucumber
>       Given The qualis "2013" has avaliations no avaliation for the journal "Nature"
>       And The researcher of cpf "11111111111" has only one article: "The Future of Computers" published at "Nature"
>       When The system create a score for the reseacher of cpf "11111111111" in the qualis "2013"
> ```

I ask the system for the score...

---

_[test/cucumber/JournalNotFound.feature, line 17 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrrGU2_dmn7AET_wNr:-KWrrGU2_dmn7AET_wNs:b-e06i6i) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L17)):_

> ``` Cucumber
>       And The researcher of cpf "11111111111" has only one article: "The Future of Computers" published at "Nature"
>       When The system create a score for the reseacher of cpf "11111111111" in the qualis "2013"
>       Then The system also creates a list of articles not found containing only "The Future of Computers"
> ```

creates -> shows? 

---

_[test/cucumber/JournalNotFound.feature, line 19 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrren9h3EGRgi8NwFP:-KWrren9h3EGRgi8NwFQ:bhbsvsq) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L19)):_

> ``` Cucumber
>       Then The system also creates a list of articles not found containing only "The Future of Computers"
> 
>     Scenario: article of a researcher not avaliated
> ```

nao entendi a diferença para o primeiro

---

_[test/cucumber/JournalNotFound.feature, line 25 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrrruwKEgWamKvMw8k:-KWrrruwKEgWamKvMw8l:b-rxsdwt) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L25)):_

> ``` Cucumber
>       Then I should see that "1" article wasn't scored
> 
>     Scenario: show page of not avaliated articles
> ```

tem erros básicos aí abaixo, como click. lê o livro. os primeiros cenários estão melhores. o texto inclusive. 

---

_[test/cucumber/steps/JournalNotFoundSteps.groovy, line 31 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrtjDswFoaqpzLEj_g:-KWrtjDswFoaqpzLEj_h:baril2o) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/steps/JournalNotFoundSteps.groovy#L31)):_

> ``` Groovy
>     avs.add(av2)
>     avs.add(av3)
>     av1.save flush:true
> ```

deve usar métodos do controlador, serviços do sistema, senão não está testando o sistema

---

_[test/cucumber/steps/JournalNotFoundSteps.groovy, line 68 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrvQrGXMteosiFzwYm:-KWrvQrGXMteosiFzwYn:b73dlg9) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/steps/JournalNotFoundSteps.groovy#L68)):_

> ``` Groovy
>     at CreateResearcherPage
>     Set<Article> h = new HashSet<Article>();
>     h.add(Article.findByJournal(journal))
> ```

esquisito passar o objeto artigo, simulando a gui teria apenas o id do artigo

---

_[test/cucumber/steps/JournalNotFoundSteps.groovy, line 74 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrve0bEKhOMY3Z1yTU:-KWrve0bEKhOMY3Z1yTV:b6vj8ax) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/steps/JournalNotFoundSteps.groovy#L74)):_

> ``` Groovy
>     to CreateQualisAvaliationPage
>     at CreateQualisAvaliationPage
>     page.CreateAvaliation(journal, "B2")
> ```

a criação do qualis é feita pela página também? não é importando um arquivo?

---

_[test/cucumber/steps/JournalNotFoundSteps.groovy, line 88 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrvoGdAwXq-gThMNXT:-KWrvoGdAwXq-gThMNXU:bswue24) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/steps/JournalNotFoundSteps.groovy#L88)):_

> ``` Groovy
> Then(~/^I should see that "([^"]*)" article wasn't scored$/) { String arg1 ->
>     at ShowResearcherScorePage
>     page.Showing()
> ```

nome de método inadequado

---

_[test/cucumber/steps/JournalNotFoundSteps.groovy, line 100 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrvvfYIvN9o5ze6l4s:-KWrvvfYIvN9o5ze6l4t:b1r9pd3) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/steps/JournalNotFoundSteps.groovy#L100)):_

> ``` Groovy
>     qualis.avaliations.each{ if(it.journal == journal) hasJournal = true}
>     assert !hasJournal
> }
> ```

repeticao de codigo

---

_[test/functional/pages/CreateArticlePage.groovy, line 5 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrwIeB0aMmEOv9vKMn:-KWrwIeB0aMmEOv9vKMo:bbj504v) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/functional/pages/CreateArticlePage.groovy#L5)):_

> ``` Groovy
> import geb.Page
> 
> class CreateArticlePage extends Page {
> ```

para que precisa disso? a criação deveria se dar através da leitura do lattes

---

_[test/functional/pages/CreateQualisPage.groovy, line 15 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrw_GG75X9pQlPOvak:-KWrw_GHJS8TUKSUTZGG:b-4cgbww) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/functional/pages/CreateQualisPage.groovy#L15)):_

> ``` Groovy
>     boolean CreateQualis(String qualisYear, Set<QualisAvaliation> avs){
>         $("form").year = qualisYear
>         $("form").avaliations = avs
> ```

passa um conjunto?

---

_[test/functional/pages/CreateResearcherPage.groovy, line 13 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrwf3ZjfjdIUJHpFtM:-KWrwf3ZjfjdIUJHpFtN:b-i2u2he) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/functional/pages/CreateResearcherPage.groovy#L13)):_

> ``` Groovy
>     }
> 
>     boolean CreateResearcher(String cpf, Set<Article> articles){
> ```

deveria ser lendo lattes

---

_[test/functional/pages/ListResearcherScorePage.groovy, line 9 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrwj-6qw6jS19JxgjR:-KWrwj-6qw6jS19JxgjS:b-85ft8l) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/functional/pages/ListResearcherScorePage.groovy#L9)):_

> ``` Groovy
> 
>     static at = {
>         title ==~ /ResearcherScore Listagem/
> ```

falta internacionalização

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pauloborba/rpa/3)_

**Comment by pauloborba** [comment]
Review status: 17 of 47 files reviewed at latest revision, 26 unresolved discussions, some commit checks failed.

---

*[test/cucumber/JournalNotFound.feature, line 32 at r6](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KX7EpsmGNqxNZ2uc3_K:-KX7EpsmGNqxNZ2uc3_L:bfi3di5) ([raw file](https://github.com/pauloborba/rpa/blob/f8385081a4e3df52a8af0fc72d1f4072bcd060ab/test/cucumber/JournalNotFound.feature#L32)):*
> ```Cucumber
>       When I ask to show the avaliation of researcher "11111151111"
>       Then I am at the description page for the avaliation
>       And I can see that "Science" journal wasn't scored
> ```

acrescenta mais um ou dois cenários detalhando melhor o comportamento

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pauloborba/rpa/3)*

**Comment by Trakton** [comment]
Review status: 17 of 47 files reviewed at latest revision, 26 unresolved discussions, some commit checks failed.

---

*[grails-app/controllers/rpa/ResearcherScoreController.groovy, line 129 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWryR3rUVKR7UcAkUcJ:-KXmWiM6POY3rxQOG6Gn:b-896fix) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/controllers/rpa/ResearcherScoreController.groovy#L129)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

merece extrair métodos e deixar a intenção de cada parte mais clara
</blockquote></details>

Done.

---

*[grails-app/domain/rpa/Article.groovy, line 6 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrygPUcz4NhZv2rFgA:-KXmWrkPRIurcnYYkq4T:b-auo0y3) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/Article.groovy#L6)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

seria importante ter os autores na próxima iteração
</blockquote></details>

Nesse caso, eu não tenho liberdade para editar essa classe, pois ela faz parte da feature de outro colega de equipe. No início do projeto, foram definidas as interfaces e atributos de cada classe básica. Talvez esse seja um ponto pra discutir com todo o time.

---

*[grails-app/domain/rpa/Article.groovy, line 14 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWryoFikCifIEBdfNJe:-KXmXC_s3DHJIoDRirsF:b-j00x01) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/Article.groovy#L14)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

nao precisa desse tipo de construtor em groovy, já é provido
</blockquote></details>

Certo. Todos os construtores desnecessários de todas as classes foram removidos!

---

*[grails-app/domain/rpa/Qualis.groovy, line 4 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrz2U6sm8JWz5JZj3i:-KXmXI6Q7BideKrrQEjv:bt0fb8u) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/Qualis.groovy#L4)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

melhor renomear para descrição na próxima iteração. ano é uma das possíveis descrições 
</blockquote></details>

Feito.

---

*[grails-app/domain/rpa/QualisEvaluation.groovy, line 3 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrzEWu6Bvq392BYBRw:-KXmXJoeBikVIOSGJa-g:bk6imsq) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/QualisEvaluation.groovy#L3)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

evaluation, não avaliation. renomear tudo
</blockquote></details>

Dei um refactor em todo  o projeto alterando todas as classes para Evaluation.

---

*[test/cucumber/JournalNotFound.feature, line 11 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrpjAO7FhPOoW1Z3S0:-KXmXSGaKB3bwwy312Ox:bt0fb8h) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L11)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

it's -> its

seria interessante colocar, para a segunda iteracao, mais Ands apos o Then verificando outros detalhes que devem aparecer além dessa mensagem
</blockquote></details>

Feito!

---

*[test/cucumber/JournalNotFound.feature, line 14 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrq75DVnEcBYp21DCO:-KXmXT_EXb2Fbdb27xqM:b-5wr3pv) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L14)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

has no evaluations 

evaluation não
</blockquote></details>

Troquei!

---

*[test/cucumber/JournalNotFound.feature, line 16 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrr1l1clVdZ20tdNCY:-KXmXUbSXmNM8nrWHq_V:b-896fix) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L16)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

I ask the system for the score...
</blockquote></details>

Done.

---

*[test/cucumber/JournalNotFound.feature, line 17 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrrGU2_dmn7AET_wNr:-KXmXVcc1FXeye-2EW0d:bb4brwl) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L17)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

creates -> shows? 
</blockquote></details>

Isso. Troquei!

---

*[test/cucumber/JournalNotFound.feature, line 19 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrren9h3EGRgi8NwFP:-KXmXaRpAK8mjRoPpe56:bmx7mi9) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L19)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

nao entendi a diferença para o primeiro
</blockquote></details>

O primeiro é focado em testar se o controller está gerando uma string que representa o score do pesquisador corretamente. Esse é pra ver se essa informação está sendo corretamente colocada na GUI.

---

*[test/cucumber/JournalNotFound.feature, line 25 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrrruwKEgWamKvMw8k:-KXmXp_PJa9l8F8539y5:b-aesh1) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L25)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

tem erros básicos aí abaixo, como click. lê o livro. os primeiros cenários estão melhores. o texto inclusive. 
</blockquote></details>

Troquei o click. Lembro que já tinha tirado essa dúvida antes. Deve ter passado despercebido.

---

*[test/cucumber/steps/JournalNotFoundSteps.groovy, line 31 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrtjDswFoaqpzLEj_g:-KXmXyOJvFR-5vhDjlB2:bouxbs6) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/steps/JournalNotFoundSteps.groovy#L31)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

deve usar métodos do controlador, serviços do sistema, senão não está testando o sistema
</blockquote></details>

Foi um erro grave. Mas o resolvi, e agora os métodos do controller estão sendo propriamente chamados.

---

*[test/cucumber/steps/JournalNotFoundSteps.groovy, line 74 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrve0bEKhOMY3Z1yTU:-KXmY6YJ0i2DfGEmGd93:boy3ffb) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/steps/JournalNotFoundSteps.groovy#L74)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

a criação do qualis é feita pela página também? não é importando um arquivo?
</blockquote></details>

Nesse caso, este é apenas um Stub criado para testar a minha feature. A parte de importar arquivo é a feature de Divino.

---

*[test/cucumber/steps/JournalNotFoundSteps.groovy, line 88 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrvoGdAwXq-gThMNXT:-KXmYCrDZdTisyXaxctN:b-64t0d0) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/steps/JournalNotFoundSteps.groovy#L88)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

nome de método inadequado
</blockquote></details>

Trocado!

---

*[test/functional/pages/CreateArticlePage.groovy, line 5 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrwIeB0aMmEOv9vKMn:-KXmYMMNsBSHeF1utjfh:bcmm8pq) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/functional/pages/CreateArticlePage.groovy#L5)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

para que precisa disso? a criação deveria se dar através da leitura do lattes
</blockquote></details>

Esse também é outro Stub apenas para possibilitar os testes da minha feature. A leitura do lattes foi feita, mas por outra pessoa.

---

*[test/functional/pages/CreateResearcherPage.groovy, line 13 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrwf3ZjfjdIUJHpFtM:-KXmYVj4FJtOz2XU8bPE:b-flsoka) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/functional/pages/CreateResearcherPage.groovy#L13)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

deveria ser lendo lattes
</blockquote></details>

Este é um stub para possibilitar os testes da minha feature. A leitura do lattes é a feature de outra pessoa.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pauloborba/rpa/3)*

**Comment by Trakton** [comment]
Review status: 17 of 50 files reviewed at latest revision, 26 unresolved discussions.

---

*[test/functional/pages/ListResearcherScorePage.groovy, line 9 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrwj-6qw6jS19JxgjR:-KXmqDCz_mGGmJ4hwa3q:b-mep4gg) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/functional/pages/ListResearcherScorePage.groovy#L9)):*
<details><summary><i>Previously, pauloborba (Paulo Borba) wrote…</i></summary><blockquote>

falta internacionalização
</blockquote></details>

Mais uma falha grave resolvida! Os testes DEVEM ser independentes de plataforma, e de linguagem. Agora sim, eles foram internacionalizados!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pauloborba/rpa/3)*

**Comment by pauloborba** [comment]
Review status: 17 of 50 files reviewed at latest revision, 17 unresolved discussions.

---

*[grails-app/controllers/rpa/ResearcherScoreController.groovy, line 129 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWryR3rUVKR7UcAkUcJ:-KY8p7X_NeRDypkEWKqf:bf42ryu) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/controllers/rpa/ResearcherScoreController.groovy#L129)):*
<details><summary><i>Previously, Trakton (Higor Cavalcanti Machado Botelho) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

intenção ainda não clara devido ao nome do método, que é pouco significativo

---

*[grails-app/domain/rpa/QualisEvaluation.groovy, line 3 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrzEWu6Bvq392BYBRw:-KY8pb8SFsyGo7zK4drh:b-8v1zp5) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/grails-app/domain/rpa/QualisEvaluation.groovy#L3)):*
<details><summary><i>Previously, Trakton (Higor Cavalcanti Machado Botelho) wrote…</i></summary><blockquote>

Dei um refactor em todo  o projeto alterando todas as classes para Evaluation.
</blockquote></details>

esqueceu logo abaixo!

---

*[test/cucumber/JournalNotFound.feature, line 11 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrpjAO7FhPOoW1Z3S0:-KY8q-LmA_tVRVXNucYw:b-qv8u3l) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/JournalNotFound.feature#L11)):*
<details><summary><i>Previously, Trakton (Higor Cavalcanti Machado Botelho) wrote…</i></summary><blockquote>

Feito!
</blockquote></details>

feito?

---

*[test/cucumber/steps/JournalNotFoundSteps.groovy, line 74 at r5](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KWrve0bEKhOMY3Z1yTU:-KY8r10hj3nVKuD0ARD4:b-pukhtd) ([raw file](https://github.com/pauloborba/rpa/blob/6313c48c7a5acc065bb4d3aef1f6ab797d60825d/test/cucumber/steps/JournalNotFoundSteps.groovy#L74)):*
<details><summary><i>Previously, Trakton (Higor Cavalcanti Machado Botelho) wrote…</i></summary><blockquote>

Nesse caso, este é apenas um Stub criado para testar a minha feature. A parte de importar arquivo é a feature de Divino.
</blockquote></details>

o stub deveria simular a importação do arquivo.

---

*[test/cucumber/steps/JournalNotFoundSteps.groovy, line 26 at r7](https://reviewable.io:443/reviews/pauloborba/rpa/3#-KY8qoGsxsAsq21K24cr:-KY8qoGsxsAsq21K24cs:b-4frcal) ([raw file](https://github.com/pauloborba/rpa/blob/62b09505d1fccfdd6b375e6c9746ac921f00107a/test/cucumber/steps/JournalNotFoundSteps.groovy#L26)):*
> ```Groovy
> ResearcherScore score
> 
> def CreateQualisEvaluation(controller, journal, evaluation){
> ```

métodos devem começar com minúsculas

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pauloborba/rpa/3)*

**Pull request closed without merging**